src/themes/default/components/core/blocks/Microcart/Microcart.vue

```diff
@@ -178,9 +178,16 @@ export default {
       this.addCouponPressed = false
     },
     setCoupon () {
-      this.applyCoupon(this.couponCode).then(() => {
+      this.applyCoupon(this.couponCode).then((response) => {
```

Comment:
to avoid code duplication here we could change method to
```js
  async setCoupon () {
      const couponApplied = await this.applyCoupon(this.couponCode)
      this.addCouponPressed = false
      this.couponCode = ''
      if (!couponApplied) {
        this.$store.dispatch('notification/spawnNotification', {
          type: 'warning',
          message: i18n.t("You've entered an incorrect coupon code. Please try again."),
          action1: { label: i18n.t('OK') }
        })
      }
    },
```

but for this we need to ensure that applyCoupon always returns boolean.